sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileClient.java

```diff
@@ -724,9 +1008,47 @@ public Response<ShareFileProperties> downloadToFileWithResponse(String downloadF
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<ShareFileProperties> downloadToFileWithResponse(String downloadFilePath, ShareFileRange range,
         ShareRequestConditions requestConditions, Duration timeout, Context context) {
-        Mono<Response<ShareFileProperties>> response = shareFileAsyncClient.downloadToFileWithResponse(downloadFilePath,
-            range, requestConditions, context);
-        return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
+        Objects.requireNonNull(downloadFilePath, "'downloadFilePath' cannot be null.");
+
+        // Setup initial conditions and file channel
+        ShareRequestConditions finalRequestConditions = requestConditions == null ? new ShareRequestConditions()
+            : requestConditions;
+        //ShareFileRange finalRange = range;
+        try (FileChannel fileChannel = FileChannel.open(Paths.get(downloadFilePath), StandardOpenOption.CREATE,
+            StandardOpenOption.WRITE)) {
+            Callable<Response<ShareFileProperties>> operation = () -> {
+                Response<ShareFileProperties> propertiesResponse = getPropertiesWithResponse(timeout, context);
+                long fileLength = propertiesResponse.getValue().getContentLength();
+                ShareFileRange finalRange = range == null ? new ShareFileRange(0, fileLength - 1) : range;
+
+                long start = finalRange.getStart();
+                long end = finalRange.getEnd();
+                long position = start;
+
+                while (position <= end) {
+                    long chunkSize = Math.min(FILE_DEFAULT_BLOCK_SIZE, end - position + 1);
```

Comment:
Another nit: Should allow for the chunk size to be configured on download. From my understanding the block size limit is only an upload thing, not a download thing.